gse_account_analytic/models/account_move.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    source = fields.Char(string="Source so", compute="_compute_source")
+
+    def _compute_source(self):
+        # compute source based on ref
+        for move in self:
+            if move.ref:
+                source = move.ref.split(" - ")[0]
+                delivery = self.env["stock.picking"].search(
+                    [("name", "=", source)], limit=1
+                )
+                so = self.env["sale.order"].search(
+                    [("name", "=", delivery.origin)], limit=1
+                )
+                move.source = so.analytic_account_id.name
+            else:
+                move.source = "Unknown"
```

Comment:
@sbuhl I've fetched directly the source inside the move.line model, using the search method. It helps us gain in efficiency because I have used indexed fields (in this case `name`), and also I have limit the parameter to `1` to limit the retrieved data.

```py
ref =  l.ref.split(" - ")[0] if l.ref else ""
related_stock_picking = self.env["stock.picking"].search([("name", "=", ref)], limit=1 )
```